### ISSUE TITLE
Include the rest of `internment`'s API

### ISSUE THREAD

**Issue by jsgf** [ryzhyk/arc-interner#4]
Even though only `ArcIntern` is flawed in `internment`, using its other APIs carries the risk that later on someone may start using `ArcIntern` without realizing its problems.

This crate should also export `Intern` and `LocalIntern` so that it can be used as a complete functional replacement.

**Comment by ryzhyk** [comment]
We should really help @droundy fix `internment` instead of stealing more of his code.  I am sure he will happily accept a PR.  I ended up forking his implementation because I do not fully understand some of the unsafe code in `internment`, but that does not mean it is in any way broken, except for the one confirmed bug (https://github.com/droundy/internment/issues/11).

For this crate, I prefer to focus on improving the `ArcInterner` functionality, including reducing contention and hashing overhead.

**Comment by droundy** [comment]
Indeed, I'd love a pull request. In a couple of weeks the quarter will be over and I will have a bit more time for fun programming. There's a backlog of things to do, but this is definitely on the list.
